apps/aehttp/src/aehttp_helpers.erl

```diff
@@ -720,6 +728,25 @@ dry_run_txs_([#{ <<"tx">> := ETx } | Txs], Acc) ->
 dry_run_txs_([#{ <<"call_req">> := CallReq } | Txs], Acc) ->
     dry_run_txs_(Txs, [{call_req, CallReq} | Acc]).
 
+
+dry_run_tx_hash(TxHash) ->
+    case aeser_api_encoder:safe_decode(tx_hash, TxHash) of
+        {ok, TxHashInternal} ->
+            case aec_chain:find_tx_with_location(TxHashInternal) of
+                {mempool, SignedTx} ->
+                    Tx = aetx_sign:tx(SignedTx),
+                    {Type, _} = aetx:specialize_type(Tx),
+                    case not lists:member(Type, [offchain_tx]) of
```

Comment:
Here you do not dry run `offchain_tx`, but when the tx is not from a hash you also exclude `paying_for_tx`, and `ga_meta_tx`.
There is a comment in the PR that you would combine the logic in one function doing the same?

Reply:
I was going to change it to a single function but executing dry runs on paying_for_tx and ga_meta_tx from mempool transactions doesn't have the same risks as running the transaction directly, so we decided to allow it.